### ISSUE TITLE
Irrigation data is not send by the land model to coupler

### ISSUE THREAD

**Issue by bishtgautam** [E3SM-Project/E3SM#2508]
The land model does not send data for irrigation to the coupler, though irrigation may be active in the land model. Absence of irrigation data in the coupler does not trigger an error (see [seq_diag_mct.F90#L868](https://github.com/E3SM-Project/E3SM/blob/master/cime/src/drivers/mct/main/seq_diag_mct.F90#L868)). Unaccounting for irrigation is expected to lead to an error in water budget diagnostic reported by coupler.

**Comment by jonbob** [comment]
@bishtgautam - is an irrigation model on for the regular B-cases, like the DECK experiments? My understanding was that it was not....

**Comment by bishtgautam** [comment]
@jonbob, I don't believe irrigation is on in the DECK experiments. Irrigation is only applied to certain vegetation types, which are not active by default. While trying to understand the water budget diagnostic in the coupler, I noticed that the irrigation flux isn't being correctly accounted for in the coupler and this issue is for fixing the coupler diagnostics when irrigiation is on in the land model.

**Comment by jonbob** [comment]
@bishtgautam - agreed. I was mostly just making sure this wasn't our smoking gun from the DECK runs
